doccano_api_client/__init__.py

```diff
@@ -33,46 +11,84 @@ class _Router:
     def get(
         self,
         endpoint: str,
+        params: dict = {},
     ) -> requests.models.Response:
         """
         Args:
             endpoint (str): An API endpoint to query.
 
         Returns:
-            requests.models.Response: The request response.
+            requests.models.Response: The request response (JSON).
+        """
+        request_url = urljoin(self.baseurl, endpoint)
+        return self._get(request_url, params=params).json()
+
+    def get_file(
+            self,
+            endpoint: str,
+            params: dict = {},
+            headers: dict = {},
+            ) -> requests.models.Response:
+        """
+        Gets a file.
         """
         request_url = urljoin(self.baseurl, endpoint)
-        return self.session.get(request_url)
+        return self._get(request_url, params=params, headers=headers)
+
+    def _get(
+            self,
+            url: str,
+            params: dict = {},
+            headers: dict = {},
+            ) -> requests.models.Response:
+        return self.session.get(url, params=params, headers=headers)
 
     def post(
-        self,
-        endpoint: str,
-        data: dict = {},
-        files: dict = {}
-    ) -> requests.models.Response:
+            self,
+            endpoint: str,
+            data: dict = {},
+            json: dict = {},
+            files: dict = {}
+            ) -> requests.models.Response:
         """
+        Used to POST arbitrary (form) data or explicit JSON.
+        Both will have the correct Content-Type header set.
         """
+        if json and data:
+            return "Error: cannot have both data and json"
+
         request_url = urljoin(self.baseurl, endpoint)
-        return self.session.post(request_url, data=data, files=files)
+        return self.session.post(
+                request_url, data=data, files=files, json=json).json()
+
+    def delete(
+            self,
+            endpoint: str,
+            ) -> requests.models.Response:
+        """
+        Deletes something at the given endpoint.
+        """
+        request_url = urljoin(self.baseurl, endpoint)
+        return self.session.delete(request_url)
 
     def build_url_parameter(
-        self,
-        url_parameter: dict
-    ) -> str:
+            self,
+            url_parameter: dict
+            ) -> str:
         """
         Format url_parameters.
-        
+
         Args:
             url_parameter (dict): Every value must be a list.
-        
+
         Returns:
             A URL parameter string. Ex: `?key1=u1&key1=u2&key2=v1&...`
         """
-        return ''.join(['?', '&'.join(['&'.join(['='.join([tup[0], str(value)]) for value in tup[1]]) for tup in url_parameter.items()])])
+        return ''.join(['?', '&'.join(['&'.join(['='.join(
```

Comment:
Consider to using `urllib.parse.urlencode`:

```python
>>> import urllib.parse
>>> d = {'key1': 'value1', 'key2': 'value2'}
>>> urllib.parse.urlencode(d)
'key1=value1&key2=value2'
```

Reply:
Actually, this part has gone untouched (apart from the _flake8_ commit) while I did consider changing it to use `requests` `params={}`. I was saving that for some other PR, to go in and fix/change current code and keep focus here around adding new code :)